core/src/main/kotlin/me/kcra/takenaka/core/mapping/resolve/impl/QuiltMappingResolver.kt

```diff
@@ -0,0 +1,269 @@
+            val checksumUrl = URL("$urlString.sha1")
+
+            if (MAPPING_JAR in workspace) {
+                checksumUrl.httpRequest {
+                    if (it.ok) {
+                        val checksum = file.getChecksum(sha1Digest)
+
+                        if (it.readText() == checksum) {
+                            logger.info { "matched checksum for cached ${version.id} Quilt Mappings" }
+                            return@resolver findMappingFile(file)
+                        }
+                    } else if (file.fileSize() == url.contentLength) {
+                        logger.info { "matched same length for cached ${version.id} Quilt Mappings" }
+                        return@resolver findMappingFile(file)
+                    }
+                }
+
+                logger.warn { "checksum/length mismatch for ${version.id} Quilt Mappings cache, fetching them again" }
+            }
+
+            url.httpRequest {
+                if (it.ok) {
+                    it.copyTo(file)
+
+                    logger.info { "fetched ${version.id} Quilt Mappings" }
+                    return@resolver findMappingFile(file)
+                }
+
+                logger.warn { "failed to fetch ${version.id} Quilt Mappings, received ${it.responseCode}" }
+            }
+
+            return@resolver null
+        }
+
+        upToDateWhen { it == null || it.isRegularFile() }
+    }
+
+    override val licenseOutput = lazyOutput<Path?> {
+        resolver {
+            val file = workspace[LICENSE]
+
+            if (LICENSE in workspace) {
+                logger.info { "found cached ${version.id} Quilt license file" }
+                return@resolver file
+            }
+
+            URL(licenseSource).httpRequest {
+                if (it.ok) {
+                    it.copyTo(file)
+
+                    logger.info { "fetched ${version.id} Quilt license file" }
+                    return@resolver file
+                } else if (it.responseCode == 404) {
+                    logger.info { "did not find ${version.id} Quilt Mappings license file" }
+                } else {
+                    logger.warn { "failed to fetch Quilt Mappings license file, received ${it.responseCode}" }
+                }
+            }
+
+            return@resolver null
+        }
+
+        upToDateWhen { it == null || it.isRegularFile() }
+    }
+
+    /**
+     * Visits the mappings to the supplied visitor.
+     *
+     * @param visitor the visitor
+     */
+    override fun accept(visitor: MappingVisitor) {
+        val mappingPath by mappingOutput
+
+        mappingPath?.reader()?.use { reader ->
+            val visitor0 = visitor.unwrap()
+
+            // FIXME: this shouldn't be here, but it's necessary for mapping-io to map Quilt parameter names
+            // add missing intermediary mappings for constructors, an equivalent of StandardProblemKinds#SPECIAL_METHOD_NOT_MAPPED
+            if (visitor0 is MappingTree) {
```

Comment:
this block can be removed after simplification to mergedv2, since we don't need to merge ourselves